tests/models/auto/test_modeling_tf_auto.py

```diff
@@ -211,6 +211,8 @@ def test_from_pretrained_with_tuple_values(self):
         config = copy.deepcopy(model.config)
         config.architectures = ["FunnelBaseModel"]
         model = TFAutoModel.from_config(config)
+        model.build()
```

Comment:
Is this related to the change ? Was it done elsewhere before (Cannot find it)

Reply:
Yes, this is related to the change! It wasn't done before and therefore was working purely out of luck.
After instantiating a TensorFlow model with `from_config`, one should build it as it otherwise has no weights attached to it.

Saving weights through the current mechanism (h5) works even with no weights attached to the model; however, that isn't the case for `safetensors` which raises an error. This caused issues when passing to `safe_serialization=True`, so I fixed the test to work correctly.

The specific error was:
```
Error
Traceback (most recent call last):
  File "/transformers/tests/models/auto/test_modeling_tf_auto.py", line 220, in test_from_pretrained_with_tuple_values
    model = TFAutoModel.from_pretrained(tmp_dir)
  File "/transformers/src/transformers/models/auto/auto_factory.py", line 565, in from_pretrained
    return model_class.from_pretrained(
  File "/transformers/src/transformers/modeling_tf_utils.py", line 2877, in from_pretrained
    with safe_open(resolved_archive_file, framework="tf") as f:
safetensors_rust.SafetensorError: Error while deserializing header: InvalidHeaderDeserialization
```